core/events.go

```diff
@@ -44,6 +44,11 @@ type ChainEvent struct {
 	FinalizedBlockHash   common.Hash
 }
 
+type Finalizer struct {
```

Comment:
I think FinalizedBlockInfo may be better name

Reply:
I do agree, fixed!